packages/wallets/src/features/cashier/modules/WithdrawalCrypto/components/WithdrawalCryptoForm/components/WithdrawalCryptoAmountConverter/WithdrawalCryptoAmountConverter.tsx

```diff
@@ -0,0 +1,141 @@
+    decimalPlacesExceeded: (limit: number) => `Up to ${limit} decimal places are allowed.`,
+    insufficientFunds: 'Insufficient funds',
+    invalidInput: 'Should be a valid number.',
+    withdrawalLimitError: (min: string, max: string, currency: string) => {
+        return `The current allowed withdraw amount is ${min} to ${max} ${currency}.`;
+    },
+};
+
+const WithdrawalCryptoAmountConverter = ({ activeWallet, exchangeRate, getCurrencyConfig }: TProps) => {
+    const [isCryptoInputActive, setIsCryptoInputActive] = useState(false);
+    const { errors, setValues, values } = useFormikContext<TForm>();
+    const FRACTIONAL_DIGITS_CRYPTO = activeWallet?.currency
+        ? getCurrencyConfig(activeWallet?.currency)?.fractional_digits
+        : 8;
+    const FRACTIONAL_DIGITS_FIAT = getCurrencyConfig('USD')?.fractional_digits;
+    const MINIMUM_WITHDRAWAL_AMOUNT = activeWallet?.currency
+        ? getCurrencyConfig(activeWallet?.currency)?.minimum_withdrawal
+        : 0;
+
+    const validateCryptoInput = (value: string) => {
+        if (!value.length) return undefined;
+
+        const amount = parseFloat(parseFloat(value).toFixed(FRACTIONAL_DIGITS_CRYPTO));
+
+        if (Number.isNaN(amount)) return helperMessageMapper.invalidInput;
+
+        if (activeWallet?.balance && amount > activeWallet?.balance) return helperMessageMapper.insufficientFunds;
+
+        if (
+            MINIMUM_WITHDRAWAL_AMOUNT &&
+            activeWallet?.balance &&
+            activeWallet.currency &&
+            amount < MINIMUM_WITHDRAWAL_AMOUNT
+        ) {
+            return helperMessageMapper.withdrawalLimitError(
+                MINIMUM_WITHDRAWAL_AMOUNT.toFixed(FRACTIONAL_DIGITS_CRYPTO),
+                activeWallet?.balance.toFixed(FRACTIONAL_DIGITS_CRYPTO),
+                activeWallet?.currency
+            );
+        }
+
+        const fractionalPart = value.split('.');
+        if (FRACTIONAL_DIGITS_CRYPTO && fractionalPart[1] && fractionalPart[1].length > FRACTIONAL_DIGITS_CRYPTO)
+            return helperMessageMapper.decimalPlacesExceeded(FRACTIONAL_DIGITS_CRYPTO);
+
+        return undefined;
+    };
+
+    const validateFiatInput = (value: string) => {
+        if (!value.length) return undefined;
+
+        if (Number.isNaN(parseFloat(value))) return helperMessageMapper.invalidInput;
+
+        const fractionalPart = value.split('.');
+        if (FRACTIONAL_DIGITS_FIAT && fractionalPart[1] && fractionalPart[1].length > FRACTIONAL_DIGITS_FIAT)
+            return helperMessageMapper.decimalPlacesExceeded(FRACTIONAL_DIGITS_FIAT);
+
+        return undefined;
+    };
+
+    return (
+        <div className='wallets-withdrawal-crypto-amount-converter'>
+            <Field name='cryptoAmount' validate={validateCryptoInput}>
+                {({ field }: FieldProps<string>) => (
+                    <WalletTextField
+                        {...field}
+                        helperMessage={errors.cryptoAmount}
```

Comment:
Set the error messages as helper message for the input upon triggering the validation function for the field.